Library/Formula/idris.rb

```diff
@@ -19,8 +18,18 @@ class Idris < Formula
   depends_on "cabal-install" => :build
   depends_on "gmp"
 
+  depends_on "libffi" => :recommended
+  depends_on "pkg-config" => :build if build.with? "libffi"
+
   def install
-    install_cabal_package
```

Comment:
Is there something we could do to make this function more useful for this case rather than having to copy-paste?